components/steering_wheel/include/IO/IO_Digital.h

```diff
@@ -0,0 +1,58 @@
+/**
+ * IO_Digital.h
+ * Digital IO Module Header
+ * @note To transfer this module to another system, the defined FLAGS and the
+ *      defined NUM_INPUTS must be modified to be congruent with the new system
+ */
+
+#pragma once
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+#include "stm32f103xb.h" /**< Needed for GPIO_TypeDef */
+#include "Types.h" 
+
+
+/******************************************************************************
+ *                              D E F I N E S
+ ******************************************************************************/
+
+#define SW1_FLAG            0x01
+#define SW2_FLAG            0x02
+#define SW4_FLAG            0x04
+#define SW5_FLAG            0x08
+#define BTN_TOP_LEFT_FLAG   0x10
+#define BTN_MIDDLE_FLAG     0x20
+#define BTN_TOP_RIGHT_FLAG  0x40
+#define NUM_INPUTS          7U /**< Must be equal to the count of flags */
+
+/******************************************************************************
+ *                             T Y P E D E F S
+ ******************************************************************************/
+
+typedef 
+#if 8 >= NUM_INPUTS
+uint8_t
+#elif 16 >= NUM_INPUTS
+uint16_t
+#elif 32 >= NUM_INPUTS
+uint32_t
+#elif 64 >= NUM_INPUTS
+uint64_t
+#endif
+IO_Digital_Flag_t; /**< Current Implementation can handle upto 64 inputs */
+
+typedef struct 
+{
+    GPIO_TypeDef*       gpiox;
+    uint16_t            pin;
+    IO_Digital_Flag_t   flag;
```

Comment:
with this strategy, if we have 64 inputs, this variable will be 64 bits long and the array that contains 64 of this struct will also contain 64x 64bit variables. This is a huge amount of memory use.

Reply:
Very strong point I had not realized, I have returned to enum type due to iterability and removed any need to store the flag